zerver/lib/actions.py

```diff
@@ -4366,10 +4366,39 @@ def do_send_create_user_group_event(user_group: UserGroup, members: List[UserPro
                  )
     send_event(event, active_user_ids(user_group.realm_id))
 
-def check_add_user_group(realm: Realm, name: Text, initial_members: List[UserProfile],
+def send_user_group_update_notification(acting_user: UserProfile, group_name: Text,
+                                        members: List[UserProfile], action: Text) -> None:
+    if action == "add_members":
+        message_template = """Hi there! We thought you'd like to know that {} ({}) \
+added you to the user group **{}**."""
+    elif action == "remove_members":
+        message_template = """Hi there! We thought you'd like to know that {} ({}) \
+removed you from the user group **{}**."""
+    elif action == "delete_group":
+        message_template = """Hi there! We thought you'd like to know that {} ({}) \
+deleted the user group **{}**."""
+
+    for member in members:
+        if member != acting_user:
+            message = message_template.format(acting_user.full_name, acting_user.email, group_name)
+            internal_send_private_message(member.realm, get_system_bot(settings.NOTIFICATION_BOT),
+                                          member, message)
+def realm_audit_log_for_user_group_creation_or_deletion(acting_user: UserProfile, user_group: UserGroup,
+                                                        action: Text) -> None:
+    extra_data = {"user_group_id": user_group.id, "user_group_name": user_group.name,
+                  "user_group_description": user_group.description}
+    RealmAuditLog.objects.create(realm=user_group.realm, acting_user=acting_user,
+                                 event_time=timezone_now(), extra_data=extra_data,
+                                 event_type=action)
+
+def check_add_user_group(created_by: UserProfile, name: Text, initial_members: List[UserProfile],
                          description: Text) -> None:
     try:
-        user_group = create_user_group(name, initial_members, realm, description=description)
+        user_group = create_user_group(name, initial_members, created_by.realm, description=description)
+        send_user_group_update_notification(created_by, user_group.name, initial_members, "add_members")
+        realm_audit_log_for_user_group_creation_or_deletion(created_by, user_group, "user_group_created")
+        realm_audit_log_for_user_group_members_update(created_by, user_group, initial_members,
+                                                      "user_added_to_group")
```

Comment:
Possibly we should have one entry for each user added?

Reply:
yes, that'd be better